src/UnigramWordPredictor.java

```diff
@@ -24,88 +25,145 @@ public UnigramWordPredictor(Tokenizer tokenizer) {
 
   /**
    * Trains the predictor using the text provided by the Scanner.
-   * The method tokenizes the text and builds a map where each word 
-   * is associated with a list of words that immediately follow it 
+   * The method tokenizes the text and builds a map where each word
+   * is associated with a list of words that immediately follow it
    * in the text. The resultant map is stored in the neighborMap
    * instance variable.
    * 
    * For example:
    * If the input text is: "The cat sat. The cat slept. The dog barked."
-   * After tokenizing, the tokens would be: ["the", "cat", "sat", ".", "the", "cat", "slept", ".", "the", "dog", "barked", "."]
+   * After tokenizing, the tokens would be: ["the", "cat", "sat", ".", "the",
+   * "cat", "slept", ".", "the", "dog", "barked", "."]
    * 
    * The resulting map (neighborMap) would be:
    * {
-   *   "the" -> ["cat", "cat", "dog"],
-   *   "cat" -> ["sat", "slept"],
-   *   "sat" -> ["."],
-   *   "." -> ["the", "the"],
-   *   "slept" -> ["."],
-   *   "dog" -> ["barked"],
-   *   "barked" -> ["."]
+   * "the" -> ["cat", "cat", "dog"],
+   * "cat" -> ["sat", "slept"],
+   * "sat" -> ["."],
+   * "." -> ["the", "the"],
+   * "slept" -> ["."],
+   * "dog" -> ["barked"],
+   * "barked" -> ["."]
    * }
    * 
    * The order of the map and the order of each list is not important.
    * 
    * @param scanner the Scanner to read the training text from
    */
+
   public void train(Scanner scanner) {
+
+    // List of words containing input from scanner
     List<String> trainingWords = tokenizer.tokenize(scanner);
+    // Create empty hashmap with String as keys and a list of strings as value
+    Map<String, List<String>> neighborMap = new HashMap<>();
 
     // TODO: Convert the trainingWords into neighborMap here
+
+    // for-loop to loop through each token. does not go to final index since it
+    // will throw an IndexOutOfBoundsExceptions
+    for (int i = 0; i < trainingWords.size() - 1; i++) {
+      // obtain the string at index i and convert to lowercase
+      String currentString = trainingWords.get(i).toLowerCase();
+      // obtain the next string after index i and convert to lowercase
+      String nextString = trainingWords.get(i + 1).toLowerCase();
+
+      // if currentString is not found within neighborMap
+      // puts currentString as key and creates a new list
+      if (!neighborMap.containsKey(currentString)) {
+        neighborMap.put(currentString, new ArrayList<>());
+      }
```

Comment:
Nice! As an alternative, can you think about how you could write this using `putIfAbsent` or `computeIfAbsent` instead?